packages/access-client/src/agent.js

```diff
@@ -409,6 +481,101 @@ export class Agent {
     }
   }
 
+  /**
+   * @param {Ucanto.DID<'key'>} space
+   * @param {Ucanto.Principal<Ucanto.DID<'mailto'>>} account
+   * @param {Ucanto.DID<'web'>} provider - e.g. 'did:web:staging.web3.storage'
+   */
+  async #addProvider(space, account, provider) {
+    return addProvider(this, space, account, provider)
+  }
+
+  /**
+   *
+   * @param {Ucanto.DID<'key'>} space
+   * @param {Ucanto.Principal<Ucanto.DID<'mailto'>>} account
+   */
+  async #delegateSpaceAccessToAccount(space, account) {
+    const issuerSaysAccountCanAdminSpace =
+      await createIssuerSaysAccountCanAdminSpace(
+        this.issuer,
+        space,
+        account,
+        undefined,
+        this.proofs([{ with: space, can: '*' }])
+      )
+    return this.invokeAndExecute(Access.delegate, {
+      audience: this.connection.id,
+      with: space,
+      expiration: Infinity,
+      nb: {
+        delegations: {
+          [issuerSaysAccountCanAdminSpace.cid.toString()]:
+            issuerSaysAccountCanAdminSpace.cid,
+        },
+      },
+      proofs: [
+        await createIssuerSaysDeviceCanAccessDelegateWithSpace(
+          this.issuer,
+          space,
+          this.issuer
+        ),
+        // must be embedded here because it's referenced by cid in .nb.delegations
+        issuerSaysAccountCanAdminSpace,
+      ],
+    })
+  }
+
+  /**
+   * Invokes voucher/redeem for the free tier, wait on the websocket for the voucher/claim and invokes it
+   *
+   * It also adds a full space delegation to the service in the voucher/claim invocation to allow for recovery
+   *
+   * @param {string} email
+   * @param {object} [opts]
+   * @param {AbortSignal} [opts.signal]
+   * @param {Ucanto.DID<'key'>} [opts.space]
+   * @param {Ucanto.DID<'web'>} [opts.provider] - provider to register - defaults to this.connection.id
+   */
+  async #newRegisterSpace(email, opts) {
```

Comment:
> Opinion alert!

Nit: I think private methods make no sense they could be separate static methods instead so not part of the interface yet usable in tests etc..

Reply:
moved to static `addProviderAndDelegateToAccount`